plugins/CoreAdminHome/OptOutManager.php

```diff
@@ -158,10 +181,324 @@ public function getQueryParameters()
     }
 
     /**
+     * Return the HTML code to be added to pages for the JavaScript opt-out
+     *
+     * @param string $piwikUrl
+     * @param string $language
+     * @param string $backgroundColor
+     * @param string $fontColor
+     * @param string $fontSize
+     * @param string $fontFamily
+     * @param bool $showIntro
+     *
+     * @return string
+     */
+    public function getOptOutJSEmbedCode(string $piwikUrl, string $language, string $backgroundColor, string $fontColor,
+                                         string $fontSize, string $fontFamily, bool $showIntro): string
+    {
+        return '<div id="matomo-opt-out"></div>
+<script src="'.$piwikUrl.'/index.php?module=CoreAdminHome&action=optOutJS&div=matomo-opt-out&language='.$language.'&backgroundColor='.$backgroundColor.'&fontColor='.$fontColor.'&fontSize='.$fontSize.'&fontFamily='.$fontFamily.'&showIntro='.$showIntro.'"></script>';
+    }
+
+    /**
+     * Return the HTML code to be added to pages for the self-contained opt-out
+     *
+     * @param string $backgroundColor
+     * @param string $fontColor
+     * @param string $fontSize
+     * @param string $fontFamily
+     * @param bool $showIntro
+     *
+     * @return string
+     */
+    public function getOptOutSelfContainedEmbedCode(string $backgroundColor, string $fontColor, string $fontSize,
+                                                    string $fontFamily, bool $showIntro): string
+    {
+        $settings = [
+            'showIntro' => $showIntro,
+            'divId' => 'matomo-opt-out',
+            'useSecureCookies' => true,
+            'cookiePath' => TrackerConfig::getConfigValue('cookie_path'),
+            'cookieDomain' => TrackerConfig::getConfigValue('cookie_domain'),
+            'cookieSameSite' => 'Lax',
+        ];
+
+        // Self contained code translations are static and always use the language of the user who generated the embed code
+        $settings = array_merge($settings, $this->getTranslations());
+        $settingsString = 'let settings = '.json_encode($settings).';';
+
+        $styleSheet = $this->optOutStyling($fontSize, $fontColor, $fontFamily, $backgroundColor, true);
+
+$code = <<<HTML
+<div id="matomo-opt-out" style=""></div>
+<script>    
+    let settings = {};         
```

Comment:
I guess we should use `var` instead of `let`, to ensure this would also work on older browsers.